compiler/include/baseAST.h

```diff
@@ -599,6 +599,7 @@ static inline const CallExpr* toConstCallExpr(const BaseAST* a)
       AST_CALL_CHILD(_a, WhileStmt,    condExprGet(),  call, __VA_ARGS__); \
                                                                            \
     } else if (isForLoop(_a)      == true) {                               \
+      AST_CALL_LIST (_a, ForLoop,      shadowVariables(), call, __VA_ARGS__); \
```

Comment:
Just noting that order of these AST* calls typically matter. There are some AST walks that start from some node and goes `->next` until it hits a marker etc. This order is a bit different than `forall`s. But you're not introducing that difference by this -- it seems to be there already.

Reply:
I want to say at some point I had this reordered but ran into use-before-def errors because of it.